### PR TITLE
ROE-1248 Update regex to match what is in the web and accounts

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-1248

Regex update and removed OWASP comment

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java [hunk 1]
```diff
@@ -45,10 +45,6 @@ public static boolean isValidCharacters(String toTest, String qualifiedFieldName
 
 
     /**
-     * The email regex is taken from OWASP https://owasp.org/www-community/OWASP_Validation_Regex_Repository
-     * ^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,15}$
-     * Sonar has highlighted issues with the * and + grouping quantifiers causing backstepping (recursion)
-     * so we are modifying them to be ++ and *+ (possessive = no backstepping)
      * @param email
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java:48]: you can remove the comments completely for isValidEmailAddress, as having the params etc on their own don't really add anything.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java:48]: actually, similar to Martin's comments in the Web changes, add a comment saying this regex is same as the ch-account service one

**Comment by mwestacott** [comment]
Do we not need corresponding unit-tests in the API (like were added in the web project) against the regular expression to test those email addresses present in the the original defect?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java:49]: Updated with the tests used in the web

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java [hunk 2]
```diff
@@ -57,8 +54,7 @@ public static boolean isValidCharacters(String toTest, String qualifiedFieldName
      */
     public static boolean isValidEmailAddress(String email, String qualifiedFieldName, Errors errs, String loggingContext) {
 
-        var regex = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*+@(?:[a-zA-Z0-9-]+\\.)++[a-zA-Z]{2,15}$";
-
+        var regex = "^.+@.+\\..+$";
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java]: Updated but presents a security issue

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java [hunk 3]
```diff
@@ -14,6 +14,9 @@ public final class StringValidators {
 
     private static final String REG_EXP_FOR_INVALID_CHARACTERS = "^[-,.:; 0-9A-Z&@$£¥€'\"«»?!/\\\\()\\[\\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zſƒǺàáâãäåāăąæǽçćĉċčþďðèéêëēĕėęěĝģğġĥħìíîïĩīĭįĵķĺļľŀłñńņňŋòóôõöøōŏőǿœŕŗřśŝşšţťŧùúûüũūŭůűųŵẁẃẅỳýŷÿźżž]*$";
 
+    //This is the same regular expression as used by the CH account service
+    private static final String REG_EXP_FOR_INVALID_EMAILS = "^.+@.+\\..+$";
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java:18]: the regex is a pattern for valid emails

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java:57]: Reverted this change

**Review by mwestacott — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**